### ISSUE TITLE
Default subject to described_class when describing a Module

### ISSUE THREAD

**Issue by postmodern** [icy-arctic-fox/spectator#6]
Noticed that `subject` defaults to `described_class.new`. This should switch to just `described_class` when `described_class.is_a?(Module)`.

```
Showing last frame. Use --error-trace for full trace.

There was a problem expanding macro 'let'

Code in macro 'subject'

 2 | let(:subject) do
     ^
Called macro defined in lib/spectator/src/spectator/dsl/values.cr:3:5

 3 | macro let(name, &block)

Which expanded to:

 > 2 | 
 > 3 |       def subject
 > 4 |         described_class.new
                               ^--
```

**Comment by icy-arctic-fox** [comment]
You are correct. It shouldn't try to use `.new` on a module. Should be fixed in 0.9.9.

Let me know if this doesn't fix it for you.

**Comment by postmodern** [comment]
Upgraded to 0.9.9 and works now. Thanks!
